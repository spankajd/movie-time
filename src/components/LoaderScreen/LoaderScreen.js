import React from 'react';
import './LoaderScreen.scss';

const LoaderScreen = ({imgPath, msg}) => {

    return msg && <div className='wrapper'>
        <img src={imgPath}/>
        <div className='message'>{msg}</div>
    </div>;
}

export default LoaderScreen;