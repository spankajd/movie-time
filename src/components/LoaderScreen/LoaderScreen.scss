
.wrapper {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    background: #DDD;
  
    img {
      background: #DDD;
      height: 150px;
    }
    .message {
      color: #ff3030;
      font-size: 24px;
      font-weight: bold;
      text-shadow: 2px 2px 4px #FFF, -2px -2px 4px #FFF;
    }
   }