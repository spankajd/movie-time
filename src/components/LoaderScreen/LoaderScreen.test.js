import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'; // Provides additional matchers
import LoaderScreen from "./LoaderScreen";

describe("LoaderScreen Component", () => {
    it("renders LoaderScreen", () => {
      render(<LoaderScreen msg="test" />);
      const element = screen.getByText(/test/i);
      expect(element).toBeInTheDocument();
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<LoaderScreen />);
        expect(asFragment()).toMatchSnapshot();
    });
});