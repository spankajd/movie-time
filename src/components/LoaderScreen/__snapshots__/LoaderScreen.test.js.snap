// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderScreen Component matches snapshot 1`] = `<DocumentFragment />`;
